pkg/serializer/serializer.go

```diff
@@ -41,6 +41,8 @@ func (w *writer) do(v reflect.Value, sliceElem bool) {
 		w.doSlice(v)
 	case reflect.Struct:
 		w.doStruct(v, sliceElem)
+	case reflect.Map:
+		w.doMap(v)
```

Comment:
This needs tests.

Reply:
What kind of tests do you mean here?

Reply:
At least the type of tests we have for other types in serializer_test.go.